src/transformers/models/decision_transformer/modeling_decision_transformer.py

```diff
@@ -830,16 +830,16 @@ def __init__(self, config):
     @replace_return_docstrings(output_type=DecisionTransformerOutput, config_class=_CONFIG_FOR_DOC)
     def forward(
         self,
-        states=None,
-        actions=None,
-        rewards=None,
-        returns_to_go=None,
-        timesteps=None,
-        attention_mask=None,
-        output_hidden_states=None,
-        output_attentions=None,
-        return_dict=None,
-    ) -> Union[Tuple, DecisionTransformerOutput]:
+        states: Optional[torch.FloatTensor] = None,
+        actions: Optional[torch.FloatTensor] = None,
+        rewards: Optional[torch.FloatTensor] = None,
+        returns_to_go: Optional[torch.FloatTensor] = None,
+        timesteps: Optional[torch.LongTensor] = None,
+        attention_mask: Optional[torch.LongTensor] = None,
```

Comment:
```suggestion
        attention_mask: Optional[torch.FloatTensor] = None,
```

Reply:
Oh, I see, they can contain nans, can't they? 
I've noticed that in the docstrings they appear as `LongTensor` (e.g. [here](https://github.com/huggingface/transformers/blob/d0cf67e6f4cfc4edb3b24e7b6446ff4f8de4ba95/src/transformers/models/decision_transformer/modeling_decision_transformer.py#L790)) so I will fix that as well but in a different commit so it could be reverted easily if you think so.

Reply:
Ah, interesting! I didn't actually notice that. In most of our other models, `attention_mask` is `FloatTensor` in both the type hints and the docstrings.

Can you test the model to see if it works with both, or if it only takes `LongTensor` / `FloatTensor`? If it accepts both, we should set the docs and type hints to use `FloatTensor` for standardization, but if it only accepts one then we should make sure the docs and hints reflect that.

Reply:
Great idea, thanks for pointing out.
The short answer is yes, they can use both:
- In Detr/Deta models, the argument is not used within the method so you can pass whatever you want
- In `DecisionTransformerModel` the [example gives](https://github.com/huggingface/transformers/blob/9ce0d5fbf288500ba1941391729da97797270164/src/transformers/models/decision_transformer/modeling_decision_transformer.py#L796) the attentions as `torch.float32` whereas [at some point later](https://github.com/huggingface/transformers/blob/9ce0d5fbf288500ba1941391729da97797270164/src/transformers/models/decision_transformer/modeling_decision_transformer.py#L796) is giving `torch.long`. This agrees with what I tested manually